politeiawww/config.go

```diff
@@ -708,36 +752,41 @@ func loadConfig() (*config.Config, []string, error) {
 				"and dbkey: %v", err)
 		}
 
-		// Validate user database encryption keys
-		cfg.EncryptionKey = util.CleanAndExpandPath(cfg.EncryptionKey)
-		cfg.OldEncryptionKey = util.CleanAndExpandPath(cfg.OldEncryptionKey)
-
-		if cfg.EncryptionKey != "" && !util.FileExists(cfg.EncryptionKey) {
-			return nil, nil, fmt.Errorf("file not found %v", cfg.EncryptionKey)
+		// Validate user database encryption keys.
+		cfg.EncryptionKey, cfg.OldEncryptionKey, err =
+			validateEncryptionKeys(cfg.EncryptionKey, cfg.OldEncryptionKey)
+		if err != nil {
+			return nil, nil, fmt.Errorf("validate encryption keys: %v", err)
 		}
 
-		if cfg.OldEncryptionKey != "" {
-			switch {
-			case cfg.EncryptionKey == "":
-				return nil, nil, fmt.Errorf("old encryption key param " +
-					"cannot be used without encryption key param")
+	case userDBMySQL:
+		// The database password is provided in an env variable.
+		cfg.DBPass = os.Getenv(envDBPass)
+		if cfg.DBPass == "" {
+			return nil, nil, fmt.Errorf("dbpass not found; you must provide " +
+				"the database password for the politeiawww user in the env " +
+				"variable DBPASS")
+		}
 
-			case cfg.EncryptionKey == cfg.OldEncryptionKey:
-				return nil, nil, fmt.Errorf("old encryption key param " +
-					"and encryption key param must be different")
+		// Validate DB host.
+		err = validateDBHost(cfg.DBHost)
```

Comment:
I'm getting a `Could not load configuration file: dbhost param is required` error.  I shouldn't get this error.  The `dbhost` param should have the correct defaults set.

If we're not doing this for cockroachdb either then it should be added as well.  This is likely my fault for not adding it previously.

Same for the encyrption key.  The encryption key is required for cockroachdb and mysql so it should default to the default location that dbutil uses which is `osDataDir/politeiawww/sbox.key`.